frb_codegen/src/commands.rs

```diff
@@ -301,6 +364,58 @@ fn guess_toolchain(dart_root: &str) -> anyhow::Result<DartToolchain> {
     Ok(DartToolchain::Dart)
 }
 
+fn ensure_dependencies(dart_root: &str, key: &str) -> anyhow::Result<Option<VersionReq>> {
+    debug!(
+        "Checking presence of {} in dependencies in {}",
+        key, dart_root
+    );
+    let manifest_file = read_file(dart_root, "pubspec.yaml")?;
+    info!("{:#?}", manifest_file);
+    let manifest_file: Pubspec = serde_yaml::from_str(&manifest_file).map_err(|e| {
+        info!("{}", e);
+        anyhow::Error::msg(format!("unable to parse pubspec.yaml in {}", dart_root))
```

Comment:
clear error handling throughout the code :)

Reply:
It should be good now.